cocotb/drivers/__init__.py

```diff
@@ -262,10 +262,12 @@ def _wait_for_signal(self, signal):
         to move to :class:`~cocotb.triggers.NextTimeStep` before
         registering more callbacks can occur.
         """
-        yield ReadOnly()
-        while signal.value.integer != 1:
-            yield RisingEdge(signal)
-            yield ReadOnly()
+        warnings.warn(
+            "Use of BusDriver._wait_for_signal() is deprecated\n"
+            "\tPlease, use cocotb.trigggers.StableValue instead",
+            DeprecationWarning, stacklevel=2
+        )
+        yield StableValue(signal, 1)
```

Comment:
The meaning changed on this line from `RisingEdge` to `Edge` - I don't think we can change things like that.

Perhaps the default should be `RisingEdge`?

Reply:
It has the same functionality. Original code used the fact that the value is 0 so waits for ``RisingEdge`` but if it would have waited for ``Edge`` it would have the same effect.
``RisingEdge`` does not work if you do ``StableValue(sig, 0)``.